spec/controllers/idv/personal_key_controller_spec.rb

```diff
@@ -148,6 +171,96 @@ def index
         expect(response).to redirect_to idv_enter_password_url
       end
     end
+
+    context 'no personal key generated yet' do
+      before do
+        idv_session.personal_key = nil
+      end
+
+      it 'generates a personal key that encrypts the idv_session profile data' do
+        get :show
+        assert_personal_key_generated_for_profile(
+          profile: idv_session.profile,
+          expected_pii: applicant,
+        )
+      end
+
+      context 'user has an existing profile in addition to the one attached to idv_session' do
```

Comment:
Can we resave their active profile with the new personal key in this case?

Reply:
Yeah we probably should!

Reply:
I added this in [e2ffbe1](https://github.com/18F/identity-idp/pull/9681/commits/e2ffbe1be733b3478a2819b62dabc07861005082), but I _also_ created [LG-11750](https://cm-jira.usa.gov/browse/LG-11750) because we need to update our personal key generation code throughout once have pii for multiple profiles in the session